tests/unit/test_log_proxy_consumer.py

```diff
@@ -0,0 +1,182 @@
+from deepdiff import DeepDiff  # type: ignore
+from ops.charm import CharmBase
+from ops.framework import StoredState
+from ops.model import Container
+from ops.testing import Harness
+
+LOG_FILES = [
+    "/var/log/apache2/access.log",
+    "/var/log/alternatives.log",
+    "/var/log/test.log",
+]
+
+HTTP_LISTEN_PORT = 9080
+GRPC_LISTEN_PORT = 9095
+
+CONFIG = {
+    "clients": [{"url": "http://10.20.30.1:3500/loki/api/v1/push"}],
+    "server": {"http_listen_port": HTTP_LISTEN_PORT, "grpc_listen_port": GRPC_LISTEN_PORT},
+    "positions": {"filename": "/opt/promtail/positions.yaml"},
+    "scrape_configs": [
+        {
+            "job_name": "system",
+            "static_configs": [
+                {
+                    "targets": ["localhost"],
+                    "labels": {
+                        "job": "juju_MODEL_123456_loki-k8s",
+                        "__path__": "/var/log/apache2/access.log",
+                    },
+                },
+                {
+                    "targets": ["localhost"],
+                    "labels": {
+                        "job": "juju_MODEL_123456_loki-k8s",
+                        "__path__": "/var/log/alternatives.log",
+                    },
+                },
+                {
+                    "targets": ["localhost"],
+                    "labels": {
+                        "job": "juju_MODEL_123456_loki-k8s",
+                        "__path__": "/var/log/test.log",
+                    },
+                },
+            ],
+        }
+    ],
+}
+
+WORKLOAD_BINARY_DIR = "/opt/promtail"
+WORKLOAD_POSITIONS_PATH = "{}/positions.yaml".format(WORKLOAD_BINARY_DIR)
+
+
+class ConsumerCharm(CharmBase):
+    _stored = StoredState()
+    metadata_yaml = textwrap.dedent(
+        """
+        name: loki-k8s
+        containers:
+          loki:
+            resource: loki-image
+          promtail:
+            resource: promtail-image
+        requires:
+          log_proxy:
+            interface: loki_push_api
+            optional: true
+        """
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args)
+        self._port = 3100
+        self._log_proxy = LogProxyConsumer(
+            charm=self, log_files=LOG_FILES, container_name="consumercharm"
+        )
+
+
+class TestLogProxyConsumer(unittest.TestCase):
+    @patch("charms.loki_k8s.v0.log_proxy.LogProxyConsumer._get_container")
+    def setUp(self, mock_container):
+        mock_container.return_value = True
+
+        self.harness = Harness(ConsumerCharm, meta=ConsumerCharm.metadata_yaml)
+        self.harness.set_model_info(name="MODEL", uuid="123456")
+        self.addCleanup(self.harness.cleanup)
+        self.harness.set_leader(True)
+        self.harness.begin()
+
+    def test__cli_args(self):
+        expected_args = "-config.file=/etc/promtail/promtail_config.yaml"
+        self.assertEqual(self.harness.charm._log_proxy._cli_args, expected_args)
```

Comment:
Not sure what is the purpose of the test - it's just str concat.
Maybe `test_cli_args_explicitly_specify_some_args()` with `self.assertIn('-config.file', ...)`?